R/summaryPopups.R

```diff
@@ -0,0 +1,74 @@
+pasteListArgumentAndItsName <- function(argumentName, list, sep) {
+  paste(argumentName, list[argumentName], sep = sep)
+}
+
+extractValuesWithNames <- function(polygons, spatialPoints, extractedValues) {
+
+  extracted <- polygons %>%
+    extract(spatialPoints)
+
+  if(!is.null(extractedValues)) {
+    extracted <- extracted[extractedValues]
+  }
+
+  names(extracted) %>% sapply(pasteListArgumentAndItsName,
+                              list = extracted, sep = ": ")
+}
+
+displayPopupWithSummary <- function(x=NULL, y=NULL, proxy, raster, polygons, rasterValueLabel = "Raster value: %s", extractedValues = NULL) {
```

Comment:
This is unnecessary. Deleted default values. Now if `x` and `y` values are missing usual missing argument error is thrown. This however should never happen in use of this function within `histogramForRaster` module, because it is triggered only when clicked.